wagtail/contrib/wagtailroutablepage/models.py

```diff
@@ -45,7 +56,7 @@ def get_subpage_urls(cls):
 
         return tuple([
             route[0]
-            for route in sorted(routes, key=lambda route: route[1])
+            for route in reversed(sorted(routes, key=lambda route: route[1]))
```

Comment:
Reversing the order breaks the current behaviour where routes defined earlier in a class will match first, e.g.:

    class RoutablePageTest(RoutablePage):
        @route(r'^archive/year/1984/$')
        def archive_for_1984(self, request):
            return HttpResponse("we were always at war with eastasia")

        @route(r'^archive/year/(\d+)/$')
        def archive_by_year(self, request, year):
            return HttpResponse("ARCHIVE BY YEAR: " + str(year))

This behaviour isn't covered by tests (or documented, I suspect...), but the `_creation_counter` logic above was designed to support this, so I think we want to preserve this behaviour. Looking at a fix now...